src/utils.jl

```diff
@@ -91,6 +91,35 @@ Converts a size given by `Tuple{N, M, ...}` into a tuple `(N, M, ...)`.
 """
 Base.@pure size_to_tuple(::Type{S}) where {S<:Tuple} = tuple(S.parameters...)
 
+@doc raw"""
+    vec2skew!(X, v, k)
+
+create a skew symmetric matrix inplace in `X` of size $k\times k$ from a vector `v`,
+for example for `v=[1,2,3]` and `k=3` this
+yields
+````julia
+[  0  1  2;
+  -1  0  3;
+  -2 -3  0
+]
+````
+"""
+function vec2skew!(X, v)
+    k = size(X)[1]
+    size(X)[2] != k && error("X is of wrong size, expected ($k,$k) got $(size(X)).")
+    n = div(k * (k - 1), 2)
+    length(v) < n && error("The vector $(v) is too short, expected $(n) got $(length(v)).")
+    m = 0
+    X .= [i < j ? (m += 1; v[m]) : zero(eltype(v)) for i in 1:k, j in 1:k]
+    X .= X - X'
+    return X
+end
+function vec2skew(v, k)
+    X = zeros(eltype(v), k, k)
```

Comment:
```suggestion
    X = similar(v, k, k)
```
I'd also like a special case for `v` being a `StaticArray` that does `similar(v, Size(k, k))` instead. And since `k` is passed as a type parameter in `Stiefel` it should probably be passed like that here as well.

Reply:
I don't understand your last sentence, for this function we do not have a manifold per se in the arguments so we ave to type to parametrize?

And – how different would the static array variant be? can you maybe help and provide it?

Reply:
I was thinking a bit more about this PR and actually that wouldn't really be a bottleneck. Let's leave this for now.

Reply:
Then I'll merge now and publish a new version.